### PR TITLE
fix: select: fixes pills, filtering, empty text, and spinner position

### PR DESCRIPTION
**Pull request by dkilgore-eightfold**
## SUMMARY:

- Adds XSmall Pill size
- Fixes Select Pills so they truncate with an ellipses and look like Pills
- Adds close button to Select Pills so they may be edited on the fly
- Adds empty state text to Select Dropdown
- Centers Select Dropdown Spinner
- Fixes right icon Text Input padding when clear button isn't shown

https://user-images.githubusercontent.com/99700808/179865168-a573931b-5241-48cb-8265-cb49b16c0cb5.mp4

## JIRA TASK (Eightfold Employees Only):
ENG-22497
ENG-22511
ENG-22510
ENG-22507

## CHANGE TYPE:

-   [X] Bugfix Pull Request
-   [ ] Feature Pull Request

## TEST COVERAGE:
TODO: Add UTs for Select and Input components

## TEST PLAN:
Pull the latest and run `yarn` and `yarn install` then verify the following components:

Select
TextInput

### CODE REVIEW

File: src/components/Pills/pills.module.scss [hunk 1]
```diff
@@ -22,11 +22,17 @@
...
-        &.small {
+        &.small,
+        &.xsmall {
             font-size: $text-font-size-2;
         }
     }
 
+    &.xsmall {
+        background: none;
+        border: none;
```
[the remark this answers is not in the document]
Comment (dkilgore-eightfold) [reply src/components/Pills/pills.module.scss:33]: Yes :)

File: src/components/Select/Select.tsx [hunk 2]
```diff
@@ -173,6 +188,25 @@ export const Select: FC<SelectProps> = React.forwardRef(
...
+            },
+            {
+                [styles.selectSize2]:
+                    size === SelectSize.Flex && smallScreenActive,
+            },
+            {
+                [styles.selectSize1]:
+                    size === SelectSize.Flex && xSmallScreenActive,
+            },
+            { [styles.selectSize1]: size === SelectSize.Large },
```
[the remark this answers is not in the document]
Comment (dkilgore-eightfold) [reply src/components/Select/Select.tsx:207]: Will update throughout the components. Yair had this idea initially, but it's confusing.

File: src/components/Select/Select.tsx [hunk 3]
```diff
@@ -11,69 +11,84 @@ import {
...
 export const Select: FC<SelectProps> = React.forwardRef(
     (
         {
-            options: _options = [],
-            inputWidth = TextInputWidth.fill,
+            classNames,
             clearable = false,
             defaultValue,
+            dropdownProps = {},
+            emptyText = 'No match found.',
```
[the remark this answers is not in the document]
Comment (dkilgore-eightfold) [reply src/components/Select/Select.tsx:35]: This particular string should be handled by the host app - ideally all strings should be translated by host. Will work with Yash over the next few sprints

File: src/components/Select/select.module.scss [hunk 4]
```diff
@@ -15,25 +20,27 @@
...
+        flex-direction: row;
         left: $space-xs;
         right: $space-xxl;
+        pointer-events: none;
+        position: absolute;
         top: 6px;
-        display: flex;
-        flex-direction: row;
-        align-items: center;
+        z-index: 1;
```
Comment (ayadav-eightfold) [inline src/components/Select/select.module.scss:31]: Let's use scss variables in this file?

Comment (dkilgore-eightfold) [reply src/components/Select/select.module.scss:31]: Will send a subsequent pr. Having issues with my fork and lfs (met my quota), need to create a new fork and clone.

**Pull request merged**